app/containers/LiquidityPage/index.js

```diff
@@ -116,54 +116,41 @@ export function LiquidityPage(props) {
     checkData()
   }, [fromAddress])
 
-  useEffect(async () => {
 
-    if (!isNotEmpty(fromSelectedToken)) {
+
+  useEffect(async () => {
+    if (!isNotEmpty(fromSelectedToken) && fromValue != '') {
       const tokenAllowance = await runApproveCheck(fromSelectedToken, wallet.address, wallet.signer);
       setFromTokenAllowance(tokenAllowance);
-      if (tokenAllowance.toString() > fromValue) {
+
+      if (Number(tokenAllowance) > Number(fromValue)) {
         setHasAllowedFromToken(true);
         setShowApprovalBox(false);
-        setOpenSupplyButton(false);
+        setOpenSupplyButton(true);
       } else {
         setHasAllowedFromToken(false);
         setShowApprovalBox(true);
         setOpenSupplyButton(false);
       }
 
     }
-    if (!isNotEmpty(toSelectedToken)) {
+    if (!isNotEmpty(toSelectedToken) && toValue != '') {
       const tokenAllowance = await runApproveCheck(toSelectedToken, wallet.address, wallet.signer);
       setToTokenAllowance(tokenAllowance);
-      if (tokenAllowance.toString() > toValue) {
+      console.log(tokenAllowance, Number(tokenAllowance) > Number(toValue))
```

Comment:
merging this with live network, you forgot to remove the consoles.

Reply:
Okay, Good catching that